### PR TITLE
8280596: ScopedMemoryAccess_closeScope: remove exception parameter

### PR DESCRIPTION
**Pull request by FrauBoes**
This change removes the trailing exception parameter in ScopedMemoryAccess_closeScope, which is now unused. 
(Plus some unrelated minimal doc cleanup that removes mentions of the old SymbolLookup class.)

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed

### Issue
 * [JDK-8280596](https://bugs.openjdk.java.net/browse/JDK-8280596): ScopedMemoryAccess_closeScope: remove exception parameter

### Reviewers
 * [Maurizio Cimadamore](https://openjdk.java.net/census#mcimadamore) (@mcimadamore - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/panama-foreign pull/637/head:pull/637` \
`$ git checkout pull/637`

Update a local copy of the PR: \
`$ git checkout pull/637` \
`$ git pull https://git.openjdk.java.net/panama-foreign pull/637/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 637`

View PR using the GUI difftool: \
`$ git pr show -t 637`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/panama-foreign/pull/637.diff">https://git.openjdk.java.net/panama-foreign/pull/637.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/prims/scopedMemoryAccess.cpp [hunk 1]
```diff
@@ -160,13 +160,12 @@ JVM_END
...
 #define MEMACCESS "ScopedMemoryAccess"
 #define SCOPE PKG_FOREIGN "ResourceScope;"
-#define SCOPED_ERR PKG_MISC MEMACCESS "$ScopedAccessError;"
 
 #define CC (char*)  /*cast a literal from (const char*)*/
 #define FN_PTR(f) CAST_FROM_FN_PTR(void*, &f)
 
 static JNINativeMethod jdk_internal_misc_ScopedMemoryAccess_methods[] = {
-    {CC "closeScope0",   CC "(" SCOPE SCOPED_ERR ")Z",           FN_PTR(ScopedMemoryAccess_closeScope)},
+    {CC "closeScope0",   CC "(" SCOPE ")Z",           FN_PTR(ScopedMemoryAccess_closeScope)},
```
Comment (mcimadamore) [inline src/hotspot/share/prims/scopedMemoryAccess.cpp:168]: Uhmm - it seems like the method declaration for `ScopedMemoryAccess_closeScope` still has a trailing exception object (further down in this file)

Comment (FrauBoes) [reply src/hotspot/share/prims/scopedMemoryAccess.cpp:166]: Oops, thanks for spotting. That's removed now.

**Review by mcimadamore — APPROVED** [review]
Looks good

**Pull request closed without merging**